neo/app/actor/RequestRouter.scala

```diff
@@ -0,0 +1,107 @@
+package actor
+
+import akka.actor.{Actor, ActorLogging, ActorRef, Props}
+import akka.pattern.{ask, pipe}
+import akka.stream.Materializer
+import akka.util.Timeout
+import edu.uci.ics.cloudberry.zion.common.Config
+import play.api.libs.json._
+import play.api.libs.ws.WSClient
+import play.api.mvc.RequestHeader
+import play.Logger
+
+import scala.concurrent.duration._
+import scala.concurrent.ExecutionContext
+
+class RequestRouter (out: ActorRef, ws: WSClient, requestHeader: RequestHeader, berryClientProp: Props, config: Config)
+                    (implicit ec: ExecutionContext, implicit val materializer: Materializer) extends Actor with ActorLogging {
+
+  import RequestRouter._
+  import KeyType._
+
+  implicit val timeout: Timeout = Timeout(20.minutes)
+  val StreamingBerryClient = context.actorOf(berryClientProp)
+  val nonStreamingBerryClient = context.actorOf(berryClientProp)
+
+  // TODO UserRequest in package model is of no use
+  override def receive: Receive = {
+    case requestBody: JsValue =>
+      (requestBody \ "transform" \\ "key")(0).asOpt[KeyType.Value] match {
+        case Some(TotalCount) =>
+          (nonStreamingBerryClient ? (requestBody \ "transform" \\ "value")(0)).mapTo[JsValue].map{
+            json => respondWrapper(json, TotalCount)
+          }.recover{
+            case e =>
+              val errorMsg = "Failed to receive response from non-streaming berryClient in totalCount query: " + e.getMessage
+              Logger.error(errorMsg)
+              respondWrapper(JsString(errorMsg), Error)
+          }.pipeTo(out)
+
+        case Some(Sample) =>
+          (nonStreamingBerryClient ? (requestBody \ "transform" \\ "value")(0)).mapTo[JsValue].map{
+            json => respondWrapper(json, Sample)
+          }.recover{
+            case e =>
+              val errorMsg = "Failed to receive response from non-streaming berryClient in Sample query: " + e.getMessage
+              Logger.error(errorMsg)
+              respondWrapper(JsString(errorMsg), Error)
+          }.pipeTo(out)
+
+        case Some(Batch) =>
+
+          // TODO need to implement
```

Comment:
This just remind me why I designed the `transformation` function in Berry!!!
In order to transfer the streaming result, we cannot use the `ask` model, since it will just receive one result and close the channel. So, instead of handling the `transformation` logic in the `router`, we have to let `Berry` transform it and directly send to the client by using the `tell` (`!`) model.